### PR TITLE
Send profile changes to keycloak

### PR DESCRIPTION
**Pull request by akikoskinen**
It's now possible to setup a Keycloak instance to receive changes, when a `Profile` is modified via a GraphQL mutation.

When a GraphQL mutation changes a `Profile`, a `profile_updated` signal is emitted. If a Keycloak instance is configured, this signal is received in a handler that sends `Profile` data to the Keycloak instance.

### CODE REVIEW

File: profiles/signals.py [hunk 1]
```diff
@@ -64,7 +65,12 @@ def profile_changes_to_keycloak(sender, instance, **kwargs):
 
     user_id = instance.user.uuid
 
-    user_data = _keycloak_admin_client.get_user(user_id)
+    try:
+        user_data = _keycloak_admin_client.get_user(user_id)
+    except HTTPError as err:
+        if err.response.status_code == 404:
```
Comment (tituomin) [inline profiles/signals.py:71]: The user not being found might indicate a problem with the overall consistency which might be good to log? I do realize that at first all users have not gone through Keycloak at all and it's completely normal, but maybe later (and even earlier) we might want to follow the amount of errors here (at least in aggregate).

**Review by tituomin — APPROVED** [review]
A couple of optional, not critical suggestions.

File: docs/config.adoc [hunk 2]
```diff
@@ -43,6 +43,13 @@ The following environment variables configure authentication in other ways:
 - `TOKEN_AUTH_AUTHSERVER_URL`: Sets the "main" authentication server's URL. The URL needs to be exactly what the authentication server reports as its `issuer` value. Default is empty.
 - `ADDITIONAL_AUTHSERVER_URLS`: Sets additional authentication server URLs as a list of strings. JWTs signed by these servers are accepted for authentication. The URLs need to be exactly what the authentication servers report as their `issuer` value. Default is empty list.
 
+It's possible to configure open-city-profile to communicate with a https://www.keycloak.org/[Keycloak] instance and use it as a user database. This does not replace the internal user database but it synchronises user data into the Keycloak instance. The user database Keycloak instance can simultaneously act as an authentication server but it doesn't have to. All the following settings are needed — if any are missing, then the user data synchronisation to Keycloak feature is disabled.
+
+- `USER_DB_KEYCLOAK_BASE_URL`: The base URL of the Keycloak server, without any path parts.
```
[the remark this answers is not in the document]
Comment (akikoskinen) [reply docs/config.adoc:48]: Or just `USER_KEYCLOAK_*`?

File: profiles/signals.py (hunk 1 above)
Comment (akikoskinen) [reply profiles/signals.py:71]: Perhaps. The problem is that logging isn't well (or at all?) established in this project so it should be first figured out what exactly to log and how.

File: docs/config.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (tituomin) [reply docs/config.adoc:48]: That sound like it's implying the keycloak somehow "belongs to" the user or is user-specific? I don't know. Just plain KEYCLOAK_* sounds better to my ears.

[the remark this answers is not in the document]
Comment (akikoskinen) [reply docs/config.adoc:48]: That could be enough. I was thinking that what if there will be different Keycloaks for different purposes. But perhaps that would be a warning sign that some design decisions should be reconsidered.

File: profiles/signals.py (hunk 1 above)
Comment (tituomin) [reply profiles/signals.py:71]: Sure! These individual logging suggestions should still be documented somewhere so I'll save a link to this conversation.

File: docs/config.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (akikoskinen) [reply docs/config.adoc]: Renamed these by removing the `USER_DB_` part.

**Pull request merged**